server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -1165,28 +1182,4 @@ public static Query createPhrasePrefixQuery(
         return spanQuery.build();
     }
 
-    @Override
-    protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
```

Comment:
was reading through the code, looks like ```doXContentBody``` in ```ParametrizedFieldMapper``` might be sufficient as it is iterating over all the params. Trying to understand why ```Builder``` has been retained in the first place if it is not access from anywhere except when ```doXContentBody``` is called which there is no need to override.

Reply:
but as per existing comment in the code and this test, can you check the history, why we might have to retain the order. Which BWC is this and how it can fail in prod.

Reply:
There is some assertion logic in [MapperService](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/MapperService.java#L272-L275) where if the index mapping versions are the same then the serialized form of the mappings should also be identical. This is where the order of the fields is important to ensure BWC with old serialization.